sympy/combinatorics/rewritingsystem_fsm.py

```diff
@@ -0,0 +1,63 @@
+class State(object):
+    '''
+    A representation of a state managed by a ``StateMachine``.
+
+    Attributes:
+        name (instance of FreeGroupElement or string) -- State name which is also assigned to the Machine.
+        transisitons (OrderedDict) -- Represents all the transitions of the state object.
+        state_type (string) -- Denotes the type (accept/start/dead) of the state.
+        rh_rule (instance of FreeGroupElement) -- right hand rule for dead state.
+    '''
+
+    def __init__(self, name, state_type=None, rh_rule=None):
+        self.name = name
+        self.transitions = {}
+        self.state_type = state_type
+        self.rh_rule = rh_rule
+        if self.state_type == 's':
+            self.state_type = "start"
+        elif self.state_type == 'a':
+            self.state_type = "accept"
+        elif self.state_type == 'd':
+            self.state_type = "dead"
```

Comment:
I think it makes more sense to favor the shorter representation. So the user may enter "start" but we should be storing it as just "s". In fact, you can just say `self.state_type = state_type[0]` and not have any of these `if` statements at all.

Reply:
I don't think it's going to be less readable with just the first letter. You could always add a comment along the lines of `state_type = 's' means a 'start' state` before the first line where you use it in other functions if you wanted.